permuta/descriptors/basis.py

```diff
@@ -96,3 +97,38 @@ def __repr__(self):
 
     def __str__(self):
         return "{{{}}}".format(", ".join(str(p) for p in self))
+
+
+class Basis(AbstractBasis):
+    """A basis class.
+
+    A PermSet can be built with a Basis instance by using the basis provided
+    to it to see if a perm should be in the PermSet or not. Additionally,
+    various fast methods exist to build a PermSet defined by a basis.
+    """
+    ALLOWED_BASIS_ELEMENT_TYPES = (Perm,)
+
+
+class MeshBasis(AbstractBasis):
+    ALLOWED_BASIS_ELEMENT_TYPES = (Perm, MeshPatt)
+
+
+def detectBasisCls(basis):
```

Comment:
Function names should be lower cased and split by underscores.

Reply:
Agree! Will fix, thanks!